lib/bn.js

```diff
@@ -3169,7 +3169,10 @@
 
   Red.prototype.imod = function imod (a) {
     if (this.prime) return this.prime.ireduce(a)._forceRed(this);
-    return a.umod(this.m)._forceRed(this);
+
+    var mod = a.umod(this.m)._forceRed(this);
+    if (mod !== a) mod.copy(a);
```

Comment:
This might be not so good performance-wise. Any ideas?

Reply:
Should we introduce `iumod` instead?

Reply:
Agree that this can introduce performance issues... I thought about new method `.move` instead `.copy`.

> Should we introduce `iumod` instead?

do you mean `BN#iumod`?

Reply:
`.move` might actually work great here!

Reply:
Do you want it as separate method or only as few lines in `Red#imod`? What about `BN#iumod`? It's doesn't cost it?

Reply:
@fanatid do you mean new method for `.move`? If so,  I think yes

Reply:
`BN#move` added in https://github.com/indutny/bn.js/pull/178/commits/fa40b8c7cf349eebffb825ad67cc554462cf1f33